athf/agents/llm/hunt_researcher.py

```
@@ -220,16 +252,46 @@ def execute(
+            if self.llm_enabled and self._llm_failures > 0:
+                if self._llm_failures >= expected_llm_skills:
+                    warnings.append(
+                        "All {} LLM research skills failed — no usable LLM "
+                        "research was produced. Every LLM-backed section "
+                        "contains an 'Error during LLM analysis' placeholder "
+                        "(the related-work section is a local similarity search "
+                        "and may still be populated). Verify the LLM provider "
+                        "is reachable and configured (ATHF_LLM_PROVIDER / API "
+                        "keys / local Ollama), then re-run.".format(
+                            expected_llm_skills,
+                        )
```

Ruff (0.16.0)

[warning] Use f-string instead of `format` call

Convert to f-string

(UP032)